api/src/main/java/org/apache/iceberg/actions/ExpireSnapshots.java

```diff
@@ -92,6 +92,16 @@
    */
   ExpireSnapshots executeDeleteWith(ExecutorService executorService);
 
+  /**
+   * Allows expiration of snapshots without any cleanup of underlying manifest or data files.
+   * <p>
+   * If {@code cleanEnabled} is set to false, data and manifest files should not been cleaned after expiration.
+   *
+   * @param cleanEnabled setting this to false will skip deleting expired manifests and files
+   * @return this for method chaining
+   */
+  ExpireSnapshots cleanExpiredFiles(boolean cleanEnabled);
```

Comment:
This action is already having `deleteWith(Consumer<String> deleteFunc);` method. 
**we can plugin dummy consumer method to avoid deleting the files.**

Default implementation is to delete files. 
https://github.com/apache/iceberg/blob/master/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java#L84-L89

Reply:
But we can have changes for call procedure I guess. We can expose an argument to `skipDelete` and use the dummy consumer in spark actions of this call procedure if the `skipDelete` = true.